lib/3scale_toolbox/remote_cache.rb

```diff
@@ -0,0 +1,153 @@
+module ThreeScaleToolbox
+  class RemoteCache < BasicObject
+
+    attr_reader :metrics_cache, :methods_cache, :backend_metrics_cache, :backend_methods_cache, :subject
+
+    def initialize(subject)
+      @subject = subject
+      # Metrics and methods cache data
+      @metrics_cache = {}
+      # methods cache data
+      @methods_cache = {}
+      # Backend Metrics and methods cache data
+      @backend_metrics_cache = {}
+      # Backend methods cache data
+      @backend_methods_cache = {}
+    end
+
+    def list_metrics(service_id)
+      return metrics_cache[service_id] if metrics_cache.has_key? service_id
+
+      subject.list_metrics(service_id).tap do |metrics|
+        metrics_cache[service_id] = metrics unless metrics.respond_to?(:has_key?) && !metrics['errors'].nil?
+      end
+    end
+
+    def list_methods(service_id, metric_id)
+      key = method_cache_key(service_id, metric_id)
+      return methods_cache[key] if methods_cache.has_key? key
+
+      subject.list_methods(service_id, metric_id).tap do |methods|
+        methods_cache[key] = methods unless methods.respond_to?(:has_key?) && !methods['errors'].nil?
+      end
+    end
+
+    def create_metric(service_id, attributes)
+      subject.create_metric(service_id, attributes).tap do |_|
+        metrics_cache.delete(service_id)
+      end
+    end
+
+    def update_metric(service_id, metric_id, attributes)
+      subject.update_metric(service_id, metric_id, attributes).tap do |_|
```

Comment:
Do we ignore errors on the possible call to update_metric?
This question applies to other similar methods

Reply:
answered